src/backward.rs

```diff
@@ -34,41 +38,48 @@ impl From<u64> for NodeType {
     }
 }
 
-pub fn json_rdb_load(rdb: *mut raw::RedisModuleIO) -> Value {
-    let node_type = raw::load_unsigned(rdb).into();
+pub fn json_rdb_load(rdb: *mut raw::RedisModuleIO) -> Result<Value, Error> {
+    let node_type = raw::load_unsigned(rdb)?.into();
     match node_type {
-        NodeType::Null => Value::Null,
+        NodeType::Null => Ok(Value::Null),
         NodeType::Boolean => {
-            let buffer = raw::load_string_buffer(rdb);
-            Value::Bool(buffer.as_ref()[0] == b'1')
+            let buffer = raw::load_string_buffer(rdb)?;
+            Ok(Value::Bool(buffer.as_ref()[0] == b'1'))
+        }
+        NodeType::Integer => {
+            let n = raw::load_signed(rdb)?;
+            Ok(Value::Number(n.into()))
+        }
+        NodeType::Number => {
+            let n = raw::load_double(rdb)?;
+            Ok(Value::Number(Number::from_f64(n).unwrap()))
         }
-        NodeType::Integer => Value::Number(raw::load_signed(rdb).into()),
-        NodeType::Number => Value::Number(Number::from_f64(raw::load_double(rdb)).unwrap()),
         NodeType::String => {
-            let buffer = raw::load_string_buffer(rdb);
-            Value::String(buffer.to_string().unwrap())
+            let buffer = raw::load_string_buffer(rdb)?;
+            Ok(Value::String(buffer.to_string().unwrap()))
```

Comment:
Is this safe? What if the buffer is not a valid UTF-8 value -- will this crash?

Reply:
I wasn't able to create JSON key with invalid UFT-8 (also tried with ReJSON 1.0.8) so was not able to create such an invalid rdb.
I changed to a more graceful failure than panicking since now we return a `Result`